Notifications: sending emails failed

Hi,

Just upgraded Graylog 4.2 to 4.3. Notifications was ok running 4.2.
Now, I get an exception and this message on web console when testing:

Error: Notification has email recipients and is triggered, but sending emails failed. null

In overview:

 An error occurred while trying to send an email! (triggered 18 minutes ago)
The Graylog server encountered an error while trying to send an email. This is the detailed error message: java.lang.NullPointerException

Exception:

```
org.graylog.scheduler.JobExecutionException: Failed permanently to execute notification, giving up - <627e625ff333e2502d11e31c/Notification_Mail/email-notification-v1>
        at org.graylog.events.notifications.EventNotificationExecutionJob.execute(EventNotificationExecutionJob.java:156) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.executeJob(JobExecutionEngine.java:172) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.lambda$handleTrigger$2(JobExecutionEngine.java:150) ~[graylog.jar:?]
        at com.codahale.metrics.Timer.time(Timer.java:151) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.handleTrigger(JobExecutionEngine.java:150) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.lambda$execute$0(JobExecutionEngine.java:119) ~[graylog.jar:?]
        at org.graylog.scheduler.worker.JobWorkerPool.lambda$execute$0(JobWorkerPool.java:121) ~[graylog.jar:?]
        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:180) [graylog.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_332]
        at com.codahale.metrics.InstrumentedThreadFactory$InstrumentedRunnable.run(InstrumentedThreadFactory.java:66) [graylog.jar:?]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_332]
Caused by: org.graylog.events.notifications.PermanentEventNotificationException: Notification has email recipients and is triggered, but sending emails failed. null
        at org.graylog.events.notifications.types.EmailEventNotification.execute(EmailEventNotification.java:93) ~[graylog.jar:?]
        at org.graylog.events.notifications.EventNotificationExecutionJob.execute(EventNotificationExecutionJob.java:135) ~[graylog.jar:?]
```

* Graylog Version: graylog-server-4.3.1-1 / raylog-integrations-plugins-4.3.1-1
* Java Version: openjdk version "1.8.0_332" 
* Elasticsearch Version: elasticsearch-oss-7.10.2-1
* MongoDB Version: mongodb-org-server-4.2.20-1
* Operating System: Rocky Linux release 8.6 (Green Obsidian)
* Browser version: Firefox 101.0.1

Email configuration:

```
server.conf:transport_email_enabled = true
server.conf:transport_email_hostname = 127.0.0.1
server.conf:transport_email_port = 25
server.conf:transport_email_use_auth = false
server.conf:transport_email_use_tls = false
server.conf:transport_email_use_ssl = false
```

Thx

--- a/graylog2-server/src/main/java/org/graylog/events/notifications/types/EmailSender.java
+++ b/graylog2-server/src/main/java/org/graylog/events/notifications/types/EmailSender.java
@@ -117,6 +117,10 @@ public class EmailSender {
             email.setFrom(config.sender());
         }
 
+        if (email.getFromAddress() == null) {
+            throw new TransportConfigurationException("No from address specified for email transport.");
+        }
+
         email.setSubject(buildSubject(config, model));
         email.addTo(emailAddress);
 

--- a/graylog2-server/src/main/java/org/graylog2/shared/email/EmailFactory.java
+++ b/graylog2-server/src/main/java/org/graylog2/shared/email/EmailFactory.java
@@ -31,6 +31,8 @@ import javax.inject.Inject;
 import javax.inject.Singleton;
 import java.util.function.Supplier;
 
+import static com.google.common.base.Strings.isNullOrEmpty;
+
 /**
  * Utility class to create preconfigured {@link Email} instances by applying the settings from
  * {@link EmailConfiguration}.
@@ -113,7 +115,9 @@ public class EmailFactory {
         email.setSSLOnConnect(configuration.isUseSsl());
         email.setStartTLSEnabled(configuration.isUseTls());
 
-        email.setFrom(configuration.getFromEmail());
+        if (!isNullOrEmpty(configuration.getFromEmail())) {
+            email.setFrom(configuration.getFromEmail());
+        }
 
         return email;
     }

